lib/bike_brigade_web/live/rider_live/show.ex

```diff
@@ -0,0 +1,71 @@
+defmodule BikeBrigadeWeb.RiderLive.Show do
+  use BikeBrigadeWeb, :live_view
+
+  import Ecto.Query, warn: false
+
+  alias BikeBrigade.Riders
+  alias BikeBrigade.Riders.Rider
+  alias BikeBrigade.Delivery.Campaign
+  alias BikeBrigade.LocalizedDateTime
+  alias BikeBrigade.Repo
+
+  @impl true
+  def mount(_params, session, socket) do
+    {:ok,
+     socket
+     |> assign(:page, :riders)}
+  end
+
+  def handle_params(%{"id" => id} = params, _url, socket) do
+    rider = Riders.get_rider!(id) |> Repo.preload(:tags)
+
+    raw_stats = stats_for(rider)
+    stats = %{
+      campaigns: campaigns(raw_stats),
+      deliveries: deliveries(raw_stats),
+      distance: distance(raw_stats),
+      latest_campaign_info: latest_campaign_info(raw_stats)
+    }
+
+    # I don't have a good datastructure for campaign history and the schedule so lets keep those just html for now
+
+    {:noreply,
+     socket
+     |> assign(:rider, rider)
+     |> assign(:stats, stats)}
+  end
+
+  defp stats_for(rider) do
+    query = from r in Rider,
+      where: r.id == ^rider.id,
+      join: t in assoc(r, :assigned_tasks),
+      join: c in assoc(t, :campaign),
+      select: %{rider_id: r.id, campaign_id: c.id, task_id: t.id, distance: t.delivery_distance}
+
+    Repo.all(query)
+  end
+
+  defp campaigns(rider_stats) do
+    rider_stats |> Enum.map(& &1.campaign_id) |> Enum.uniq() |> Enum.count
+  end
+
+  defp deliveries(rider_stats) do
+    rider_stats |> Enum.map(& &1.task_id) |> Enum.uniq() |> Enum.count
+  end
+
+  defp distance(rider_stats) do
+    rider_stats |> Enum.map(& &1.distance) |> Enum.sum()
+  end
+
+  defp latest_campaign_info(rider_stats) do
+    latest_campaign_id = rider_stats |> Enum.map(& &1.campaign_id) |> Enum.max(fn -> nil end)
```

Comment:
Using the max ID is a bit icky, maybe something like?

```
campaign_ids = rider_stats |> Enum.map(& &1.campaign_id)
query = Ecto.Query.from(c in Campaign, where: c.id in ^campaign_ids, order_by: [desc: c.delivery_start])
```

Though the whole `rider_stats` query is maybe a bit weird too.

Reply:
On second thought check out https://github.com/bikebrigade/dispatch/pull/36/files